### PR TITLE
Fix the curl static build

### PR DESCRIPTION
**Pull request by donny-dont**
Fixed all package warnings for the port. Encorporated some of the vcpkg curl port to ensure a successful static build.

### CODE REVIEW

File: ports/curl/patches/0002-Dont-set-_USRDLL-on-a-static-Windows-build.patch [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
+index 2575288f7091..46973bf2b875 100644
+--- a/lib/CMakeLists.txt
++++ b/lib/CMakeLists.txt
+@@ -95,10 +95,6 @@ endif()
+ 
+ target_link_libraries(${LIB_NAME} ${CURL_LIBS})
+ 
+-if(WIN32)
```
Comment (foopoiuyt) [inline ports/curl/patches/0002-Dont-set-_USRDLL-on-a-static-Windows-build.patch:18]: This seems to be different than what I see in https://github.com/microsoft/vcpkg/blob/master/ports/curl/0011_fix_static_build.patch assuming that's intended to be the same thing.

Is there a good reason to do it differently? It seems like it might make following future changes harder if we go our own way.

File: ports/curl/portfile.cmake [hunk 2]
```diff
@@ -29,7 +31,6 @@ set(BUILD_OPTIONS
     # CMAKE options
     -DCMAKE_USE_GSSAPI=OFF
     -DCMAKE_USE_LIBSSH2=OFF
-    -DCMAKE_USE_OPENLDAP=OFF
```
Comment (foopoiuyt) [inline ports/curl/portfile.cmake:32]: Is this related to static builds or is it a drive-by fix?

Comment (donny-dont) [reply ports/curl/portfile.cmake:32]: It was a warning saying it was unused when doing a build. Can move to a separate commit.

File: ports/curl/patches/0002-Dont-set-_USRDLL-on-a-static-Windows-build.patch (hunk 1 above)
Comment (donny-dont) [reply ports/curl/patches/0002-Dont-set-_USRDLL-on-a-static-Windows-build.patch:18]: I landed the fix in https://github.com/curl/curl/pull/8030 so it will end up there in the next release

**Review by foopoiuyt — APPROVED** [review]
